### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by GTorlai**
This PR implemented the following changes:
- Change definition of `Rz`/`CRz` rotations, added `CPHASE` gate and `P(phi)` gate #254 
- Fix an issue in building the Choi matrix, where it was only possible to build the Choi matrix for a unitary circuit and a noise model as input parameters, while not for a circuit data structure already containing tuples for Kraus operators.
- Fix observer time keeping in tomography
- Added `expect` for `ITensor`

### CODE REVIEW

File: src/circuits/gates.jl [hunk 1]
```diff
@@ -73,14 +73,15 @@ gate(::GateName"H") = [
...
-  0 im
+  0 exp(im*ϕ)
 ]
 
-gate(::GateName"P") = gate("Phase")
+gate(::GateName"P"; kwargs...)     = gate("Phase"; kwargs...)
+gate(::GateName"PHASE"; kwargs...) = gate("Phase"; kwargs...)
 
-gate(::GateName"S") = gate("Phase")
+gate(::GateName"S"; kwargs...) = gate("Phase"; kwargs...)
```
Comment (mtfishman) [inline src/circuits/gates.jl:84]: Should we define this as:
```julia
gate(::GateName"S") = gate("Phase"; ϕ=π/2)
```
to match the Wikipedia definition? (https://en.wikipedia.org/wiki/Quantum_logic_gate#Phase_shift_gates)

File: src/measurements.jl [hunk 2]
```diff
@@ -189,3 +33,37 @@ end
 function entanglemententropy(ρ0::LPDO; kwargs...)
   return error("Measurement of entanglement entropy for LPDOs not yet implemented")
 end
+
+function expect(T₀::ITensor, ops::AbstractString...; kwargs...)
```
Comment (mtfishman) [inline src/measurements.jl:37]: That's a great idea to have this. We should consider moving it to ITensors.jl but that is not urgent.

File: src/circuits/gates.jl (hunk 1 above)
Comment (GTorlai) [reply src/circuits/gates.jl:84]: Good catch

File: src/measurements.jl (hunk 2 above)
Comment (GTorlai) [reply src/measurements.jl:37]: I will add this together with the MPO version when ready

**Pull request merged**